### DISCUSSION TITLE
Private channels?

### DISCUSSION THREAD

**Discussion by Fliper05** [saleor/saleor#10204] [Q&A] [answered]
Hi there!
I was wondering if it is possible to create a channel that only certain people have access to. I want to create a special price for long-term customers with different sorts of services/shipping rates. But the reason why I ask about this in the first place is that if let's say a normal customer were to find out about this, they would have access to experimental/things for long-term customers, which is made available to everyone, would not be very viable. I remember reading about 2 different store usage through one database, which could be this use case. That's why I am wondering if it would even be possible. 

Your help and guidance on how to achieve this is greatly appreciated

**Comment by maarcingebala** [comment, marked as answer]
Hi @Fliper05, currently, there is no way to make a channel private, although we see a lot of use-cases for such a feature and will probably add it in the future. For now, I see two potential solutions for you. Since the `channels` query requires admin access, storefront users cannot list available channels. Also, to access the channel's catalog, the storefront needs to know the channel slug.

Solution 1:
Create a channel for your special offer and a dedicated storefront that would require authentication and would have this channel's slug hardcoded. In the storefront, you would use the channel's slug to display products from the particular catalog. The tradeoff here is that the slug would be available in your source code, so it potentially could be found by someone and used to query the catalog through API. The question is if you bother about such risk. Not a perfect solution, but could work for some.

Solution 2:
Assign the "private" channel's slug to metadata of users, who should get access to it. Then in the storefront, when a user logs in, you could read the slug and use it to query the special catalog. The slug should be hard to guess, e.g. a UUID or some other random code. This solution would require some manual work, but maybe it would work in your use case.
